alacritty/src/display/hint.rs

```diff
@@ -306,13 +306,16 @@ struct HintPostProcessor<'a, T> {
 impl<'a, T> HintPostProcessor<'a, T> {
     /// Create a new iterator for an unprocessed match.
     fn new(term: &'a Term<T>, regex: &'a RegexSearch, regex_match: Match) -> Self {
-        let end = *regex_match.end();
-        let mut post_processor = Self { next_match: None, start: end, end, term, regex };
+        let mut post_processor = Self {
+            next_match: None,
+            start: *regex_match.start(),
+            end: *regex_match.end(),
+            term,
+            regex,
+        };
 
         // Post-process the first hint match.
-        let next_match = post_processor.hint_post_processing(&regex_match);
-        post_processor.start = next_match.end().add(term, Boundary::Grid, 1);
-        post_processor.next_match = Some(next_match);
+        post_processor.find_next_match(regex_match);
```

Comment:
The hint post processor is an iterator over all processed matches, so if the post processing results in an empty match, there should not be any other match returned by the post-processor. It's not a regex match iterator.

Reply:
OK, if I'm reading this correctly, then if post-processing returns None, there shouldn't be any other results from this iterator. That has me thinking that the other comment of
> The next match would still need to be post-processed too.

doesn't really apply because there shouldn't be anymore results since post-processing resulted in `None`.

Reply:
The post-processing resulted in `None`, but when you call the post-processor from inside the regex match iterator you still have to go to the next match and post-process that. You can't just stop without any further matches just because your first match is empty.

Now you might not need to do anything to achieve this, but you can't ever just call `regex_search_right` directly without post-processing it first since every match returned by `regex_search_right` might itself contain multiple other matches.